### PR TITLE
Add segmenter be 

### PR DESCRIPTION
**Pull request by ekellbuch**
This change Includes a new backbone (vitBE) and a new decoder  (linear+BE) for the segmenter vit model.

This change temporarily removes stochastic depth regularization (stochastic_depth flag) in segmenter to facilitate the use of vit.py and vit_batchensemble.py blocks when building new models.

### CODE REVIEW

File: uncertainty_baselines/models/segmenter_be.py [hunk 1]
```diff
@@ -0,0 +1,211 @@
...
+          num_heads=self.backbone_configs.num_heads,
+          patches=self.patches,
+          hidden_size=self.backbone_configs.hidden_size,
+          dropout_rate=self.backbone_configs.dropout_rate,
+          attention_dropout_rate=self.backbone_configs.attention_dropout_rate,
+          classifier=self.backbone_configs.classifier,
+          name='backbone')(
+              x, train=train)
+    elif self.backbone_configs.type == 'vit_be':
+      # import pdb; pdb.set_trace()
```
Comment (dustinvtran) [inline uncertainty_baselines/models/segmenter_be.py:142]: nit: rm

File: uncertainty_baselines/models/segmenter_be.py [hunk 2]
```diff
@@ -0,0 +1,211 @@
...
+    elif self.decoder_configs.type == 'linear_be':
+      output_projection = ed.nn.DenseBatchEnsemble(
+          self.num_classes,
+          self.backbone_configs.ens_size,
+          activation=None,
+          alpha_init=ed.nn.utils.make_sign_initializer(
+              self.backbone_configs.get("random_sign_init")),
+          gamma_init=ed.nn.utils.make_sign_initializer(
+              self.backbone_configs.get("random_sign_init")),
+          kernel_init=self.head_kernel_init,
```
Comment (dustinvtran) [inline uncertainty_baselines/models/segmenter_be.py:174]: `kernel_init` should be consistent with linear case, so you can do just the zeros here or set it as a config in both.

File: uncertainty_baselines/models/segmenter_be.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+    new_input_shape = tuple([input_shape[0]*ens_size,] + list(input_shape[1:-1]))
+    assert new_input_shape == x.shape[:-1], (
+        'BE Input and output shapes do not match: %d vs. %d.', new_input_shape,
+        x.shape[:-1])
+
+    return x, out
+
+
+def segmenter_be_transformer(num_classes: int,
```
Comment (dustinvtran) [inline uncertainty_baselines/models/segmenter_be.py:199]: We've actually been moving to using just the class implementations, without the need for the functional alias. How about importing SegVitBE in `__init__.py`? You could call it, say, `SegmenterBE`. Similar for `segmenter.py`.

File: uncertainty_baselines/models/segmenter_be_test.py [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+
+class SegVitBETest(parameterized.TestCase):
+
+  @parameterized.parameters(
+    (2, 16, 224, 224, 'vit', 'linear', 1),
+    (2, 16, 224, 224, 'vit_be', 'linear', 3),
+    (2, 16, 224, 224, 'vit_be', 'linear_be', 3),
+  )
+  def test_segmenter_be_transformer(self, num_classes, hidden_size, img_h, img_w,
+                                 encoder_type, decoder_type, ens_size):
```
Comment (dustinvtran) [inline uncertainty_baselines/models/segmenter_be_test.py:33]: Needs proper indent.

**Review by dustinvtran — CHANGES_REQUESTED** [review]
Thanks Kelly. Added some comments.

**Comment by ekellbuch** [comment]
Thanks for the comments @dustinvtran.

File: uncertainty_baselines/models/segmenter_be.py (hunk 3 above)
Comment (ekellbuch) [reply uncertainty_baselines/models/segmenter_be.py]: I changed this but I was wondering why we used functional aliases before and not now? thank you!

Comment (dustinvtran) [reply uncertainty_baselines/models/segmenter_be.py]: The original motivation was to have consistency where some models are Python functions that build Keras models through their functional API (ex: https://github.com/google/uncertainty-baselines/blob/main/uncertainty_baselines/models/wide_resnet.py#L144). I think that consistency resulted in more confusion in clarity though.

**Review by dustinvtran — APPROVED** [review]

**Pull request merged**